### DISCUSSION TITLE
Why this addon hasnt got into Blender 3d yet?

### DISCUSSION THREAD

**Discussion by Rockbard** [mrachinskiy/commotion#20] [General]
Kudos to the creators of this genius addon. 
 Maybe some request should be done for this addon getting into the master branch? 

Thanks!

**Comment by mrachinskiy** [comment]
Thank you! I don't know about genius, but it is good to know my tools are helpful.

Commotion is not shipped with Blender because I don't see any benefits to that, but here are few counterarguments:

- Currently Blende ships with 140 add-ons, Commotion will just be yet another add-on and most likely will not reach new users.
- I will have to comply with Blender code and UI guidelines. For example import/export add-ons required to export the whole scene by default, which I and many other users find obnoxious, but that is a guideline.

The only positive side to that I can think of, is that users wouldn't have to download and install the add-on, but since Commotion has auto-update functionality they have to do it only once.

**Comment by Rockbard** [comment]
Oh, it's definitely useful. You should contact the Blender guys I think, Maybe the changes wouldn't be so drastic as you expect, but the addon will get a lot more noticeable in the community.  The offset from the 3d cursor is made me happy when I saw it.
 
  Also since we already speaking about this addon. I just started to use it, but I already see a little hickup, How do you think, can we include an active selection for the keys. For example, I want to copy and then offset only the selected keys, right now I can not do it. 
  
  And a broad suggestion for future releases. I missing the animation presets from the After Effects a lot. 
  And, the Duik and Motion 2 script for AE had these sliders and buttons for the speed value of the keys, that's would be useful as well.
  
  But again, beautiful addon, a genius idea. Thank you so much and good luck with future endeavors.

**Comment by mrachinskiy** [comment]
> Maybe the changes wouldn't be so drastic as you expect

I am a maintainer for multiple bundled add-ons, I know exactly what to expect.

> For example, I want to copy and then offset only the selected keys, right now I can not do it.

Copying individual keyframes is not supported, and at the moment I have no plans to implement it.

> I missing the animation presets from the After Effects a lot.
> Duik and Motion 2 script for AE had these sliders and buttons for the speed value of the keys, that's would be useful as well.

It is better to give exact use cases for proposed features, I have never used AE.

**Comment by Rockbard** [comment]
Oh, I see. Ok, thanks for the quick replies.
